app/models/concerns/user_permissions.rb

```diff
@@ -72,6 +98,17 @@ def list_permissions(list)
 
     private
 
+    # ACCESS RULE HELPER
+    def update_permission(resource_type, access_rules, operation)
+      permission = @user.user_permissions.find_or_create_by(resource_type: resource_type.to_s)
+      klass = "UserPermissions::#{resource_type}AccessRules".constantize
+      new_access_rules = klass.update(permission.access_rules, access_rules, operation).to_json
```

Comment:
I think some comments or refactoring here might help explain what's going on, and/or moving some logic to the active record model UserPermission.

This makes two similar #update calls: one is klass.update (which at this time is always TagAccessRules) and Permission.update (which executes sql via the @user.user_permissions model). Perhaps this entire helper function can be moved to the UserPermission model?

Reply:
as discussed, i think this is clumsy and that there's a neat refactor to move `#update` onto the `UserPermission` model (which would already know its own `resource_type` and could statically enumerate its own `access_rule` logic sets), but we think we should put off this refactor (1) so we can keep moving, (2) b/c we are not touching tag permissions anytime soon, (3) b/c this refactor properly belongs within a larger refactor that moves more logic to `UserPermission` and (4) we want to handle permissions refactoring in one fell swoop as part of cleaning up after killing symphony.